### PR TITLE
Swc 1069

### PR DESCRIPTION
**Pull request by dburdick**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -304,7 +304,7 @@
 		<dependency>
 			<groupId>org.sagebionetworks</groupId>
 			<artifactId>markdown</artifactId>
-			<version>2014-04-17-e3f6206-7</version>
+			<version>2014-04-01-d50e530-6</version>
```
Comment (jay-hodgson) [inline pom.xml]: revert?

Comment (jay-hodgson) [inline pom.xml]: Needs to be merged with upstream

File: src/main/java/org/sagebionetworks/web/client/DisplayUtils.java [hunk 2]
```diff
@@ -1704,6 +1705,26 @@ public static String createFileEntityUrl(String baseFileHandleUrl, String entity
...
+
+	/**
+	 * Create the url to a Table cell file handle.
+	 * @param baseFileHandleUrl
+	 * @param details
+	 * @param preview
+	 * @param proxy
+	 * @return
+	 */
+	public static String createTableCellFileEntityUrl(String baseFileHandleUrl, TableCellFileHandle details, boolean preview, boolean proxy){
```
Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/client/DisplayUtils.java:1717]: FileHandle not FileEntity

Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/client/DisplayUtils.java:1717]: and remove nocache

File: src/main/java/org/sagebionetworks/web/client/DisplayUtils.java [hunk 3]
```diff
@@ -2207,4 +2217,15 @@ public static String getInfoHtml(String safeHtmlMessage) {
 		 return "<div class=\"alert alert-info\">"+safeHtmlMessage+"</div>";
 	 }
 
+	public static String getTableRowViewAreaToken(String id) {
```
Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/client/DisplayUtils.java:2220]: lets move this to the Synapse place

File: src/main/java/org/sagebionetworks/web/client/widget/table/FileHandleCell.java [hunk 4]
```diff
@@ -16,24 +26,23 @@
...
+	SynapseJSNIUtils synapseJSNIUtils;
+	PortalGinInjector ginInjector;
+	Uploader uploader;
+	
     /**
      * The HTML templates used to render the cell.
      */
     interface Templates extends SafeHtmlTemplates {
-    	@SafeHtmlTemplates.Template("<a href=\"{0}\" target=\"_blank\"><img class=\"margin-top-5\" src=\"{1}\"  height=\"100%\" width=\"100%\" /><br/>{2}</a>") 
+    	@SafeHtmlTemplates.Template("<a href=\"{0}\" target=\"_blank\"><img class=\"margin-top-5\" src=\"{1}\"  style=\"max-height:150px; \"/></a><br/><a class=\"btn btn-default btn-sm margin-top-5 margin-right-5\" href=\"{0}\" target=\"_blank\">{2}</a>")
```
Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/client/widget/table/FileHandleCell.java:44]: also add css for 100% dynamic resizing?

File: src/main/java/org/sagebionetworks/web/client/widget/table/TableModel.java [hunk 5]
```diff
@@ -46,7 +48,16 @@ public void setId(String id) {
 	}
 	
 	public String getId() {
+		if(id.startsWith(TEMP_ID_PREFIX)) return null;
 		return id;
+	}	
+	
+	public void setVersionNumber(String versionNumber) {
```
Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/client/widget/table/TableModel.java:55]: remove set

File: src/main/java/org/sagebionetworks/web/client/widget/table/TableViewUtils.java [hunk 6]
```diff
@@ -413,16 +419,42 @@ public void onFailure(Throwable caught) {
...
+								rowUpdater.updateRow(object, new AsyncCallback<RowReferenceSet>() {								
+									@Override
+									public void onSuccess(RowReferenceSet result) { 
+										updateRowIdAndVersion(object, result, view);										
+										cellTable.redrawRow(index); // to get updated URLs												
+									}
+									
+									@Override
+									public void onFailure(Throwable caught) {
+										cellTable.redraw();
```
Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/client/widget/table/TableViewUtils.java:449]: restore original file handle id

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 7]
```diff
@@ -2689,5 +2688,39 @@ public String deleteApiKey() throws RestServiceException {
...
+		org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+		try {
+			RowReferenceSet toDeleteSet = new RowReferenceSet(adapterFactory.createNew(toDelete));
+			RowReferenceSet responseSet = synapseClient.deleteRowsFromTable(toDeleteSet);			
+			return responseSet.writeToJSONObject(adapterFactory.createNew()).toJSONString();
+		} catch (SynapseTableUnavailableException e) {
+			try {
+				throw new TableUnavilableException(e.getStatus().writeToJSONObject(adapterFactory.createNew()).toJSONString());
+			} catch (JSONObjectAdapterException e1) {
+				throw new TableUnavilableException(e.getMessage());
```
Comment (dburdick) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java:2703]: rename to TableUnavailableException!

**Pull request merged**